File: templates/java/zh/text/code/moveToWall.java
/*
 * 文件：移动到墙.java
 * -----------
 * 使用“while”循环到move 卡雷尔，直到
 * 撞到墙壁。适用于任何规模的世界。
 */
import stanford.卡雷尔.*;
public class Move到墙 extends 卡雷尔 {
   public void run() {
      move到墙();
   }

   /* 这是一个非常有用的方法 */
   private void move到墙() {
      while(frontIsClear()) {
         move();
      }
   } 
}

File: templates/python/zh/text/code/moveToWall.py
＃文件：MoveToWall.py
＃------------------------------
＃使用“ while”循环到move卡雷尔，直到命中
＃ 一堵墙。适用于任何大小的世界。
from 卡雷尔.stanford import *

＃程序以main
def main():
   ＃调用move to wall函数
   move至墙()

＃这是一个非常有用的功能
def move至墙():
   ＃在条件成立时重复身体
   while front_is_clear():
      move()


